exp/term/ansi/wrap.go

```diff
@@ -336,27 +344,32 @@ func Wrap(s string, limit int, breakpoints string) string {
 				addNewline()
 			case unicode.IsSpace(r):
 				addWord()
-				space.WriteByte(b[i])
+				space.WriteRune(r)
+			case r == '-':
+				fallthrough
 			case runeContainsAny(r, breakpoints):
 				addSpace()
-				addWord()
-				buf.WriteByte(b[i])
-				curWidth++
-			default:
-				if wordLen+1 > limit {
+				if curWidth+wordLen+1 > limit {
```

Comment:
Consider dropping the `+1` and making it a `>=` comparison to simplify the logic?